content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/Methods.class/__methods__/available_credentials.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Description: provide the dynamic list content from available credentials
+# for Ansible Provider
+#
+module ManageIQ
+  module Automate
+    module AutomationManagement
+      module AnsibleTower
+        module Operations
+          class AvailableCredentials
+            def initialize(handle = $evm)
+              @handle = handle
+            end
+
+            def main
+              fill_dialog_field(fetch_list_data)
+            end
+
+            private
+
+            def provider
+              # for provisioning we use     service_template
+              # for reconfig and retire use service
+              service = @handle.root['service_template'] || @handle.root['service']
+              service.try(:job_template, 'Provision').try(:manager)
+            end
+
+            def credentials
+              result = provider.try(:credentials) || []
+              result.select { |c| c.type == @handle.inputs['credential_type'] }
```

Comment:
@bzwei 
The credential_type is set as an input parameter into the method. So the 2 instances **available_network_credential** will set 
credential_type =
ManageIQ::Providers::AnsibleTower::AutomationManager::NetworkCredential
**available_machine_credentials** will set 
credential_type =
ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential

The method itself has a default credential_type set to
ManageIQ::Providers::AnsibleTower::AutomationManager::MachineCredential